src/nupic/algorithms/Cells4.cpp

```diff
@@ -2827,13 +2816,14 @@ void Cells4::computeForwardPropagation(CState& state)
 {
   // Zero out previous values
   // Using memset is quite a bit faster on laptops, but has almost no effect
-  // on Neo15!
+  // on Neo15! // TODO shouldn't we optimize for generic (laptop) CPUs rather than
+  // a cellphone CPU
   _inferActivity.reset();
 
   // Compute cell and segment activity by following forward propagation
   // links from each source cell.  _cellActivity will be set to the total
   // activity coming into a cell.
-#if defined(NTA_ARCH_32) && defined(NTA_OS_DARWIN)
+#ifdef NTA_ARCH_64
```

Comment:
You changed the behavior here. Why do we want to switch this from 32 bit OS X to all 64 bit platforms?

Also, the comment for the endif doesn't match.